sql/core/src/main/scala/org/apache/spark/sql/execution/adaptive/ShufflePartitionsUtil.scala

```diff
@@ -296,4 +296,63 @@ object ShufflePartitionsUtil extends Logging {
     tryMergePartitions()
     partitionStartIndices.toArray
   }
+
+  /**
```

Comment:
Are these new methods exactly the same as the removed ones in `OptimizeSkewedJoin`?

Reply:
and do all the methods need to be public?

Reply:
they are same with a little change `ShufflePartitionsUtil.splitSizeListByTargetSize` -> `splitSizeListByTargetSize`. 

updated the method scope